### PR TITLE
Paths dependencies and Nginx configuration 

### PR DESCRIPTION
**Pull request by sebdevaux**
**Issue**

https://gravitee.atlassian.net/browse/APIM-438
https://github.com/gravitee-io/issues/issues/8728

**Description**

Auto-generate environment variables `PORTAL_BASE_HREF` and `CONSOLE_BASE_HREF` when they are needed and not provided in `env`

Remove the nginx annotation when the ingress class is not nginx

@brasseld FYI

### CODE REVIEW

File: apim/3.x/templates/_helpers.tpl [hunk 1]
```diff
@@ -299,3 +299,37 @@ Usage:
...
+{{- print "false" -}}
+{{- end }}
+{{- end }}
+
+{{- define "portal.base_href" -}}
+{{- if eq .Values.portal.ingress.path "/" }}
+{{- print "/" -}}
+{{ else}}
+{{- $name := regexFind "\\w+" .Values.portal.ingress.path -}}
+{{- print "/" $name -}}
```
Comment (gaetanmaisse) [inline apim/3.x/templates/_helpers.tpl:337]: No trailing "/" on that one? 

I mean in comparison of `{{- print "/" $name "/" -}}` L316

File: apim/3.x/tests/portal/deployment_test.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - hasDocuments:
+          count: 1
+      - equal:
+          path: spec.template.spec.containers[0].env[1].name
+          value: "PORTAL_BASE_HREF"
+      - equal:
+          path: spec.template.spec.containers[0].env[1].value
+          value: "/"
+
+  - it: Should generate a CONSOLE_BASE_HREF environment variable when modifying ingress path
```
Comment (gaetanmaisse) [inline apim/3.x/tests/portal/deployment_test.yaml:18]:
```suggestion
  - it: Should generate a PORTAL_BASE_HREF environment variable when modifying ingress path
```

File: apim/3.x/tests/portal/deployment_test.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - hasDocuments:
+          count: 1
+      - equal:
+          path: spec.template.spec.containers[0].env[1].name
+          value: "PORTAL_BASE_HREF"
+      - equal:
+          path: spec.template.spec.containers[0].env[1].value
+          value: "/portal"
+
+  - it: Should override generated CONSOLE_BASE_HREF environment variable
```
Comment (gaetanmaisse) [inline apim/3.x/tests/portal/deployment_test.yaml:34]:
```suggestion
  - it: Should override generated PORTAL_BASE_HREF environment variable
```

File: apim/3.x/values-seb.yaml
Comment (gaetanmaisse) [inline apim/3.x/values-seb.yaml:1]: 🤔

File: am/Chart.yaml [hunk 4]
```diff
@@ -23,4 +23,5 @@ annotations:
   # List of changes for the release in artifacthub.io
   # https://artifacthub.io/packages/helm/graviteeio/am?modal=changelog
   artifacthub.io/changes: |
-    - Add hook-delete-policy
\ No newline at end of file
+    - Add hook-delete-policy
+    - remove ingress nginx annotation when `ingress.class` in not nginx
```
Comment (gaetanmaisse) [inline am/Chart.yaml:27]:
```suggestion
    - Remove ingress nginx annotation when `ingress.class` is not nginx
```

File: am/CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,7 @@ This file documents all notable changes to [Gravitee.io Access Management 3.x](h
 ### 1.0.57
 
 - [X] Add hook-delete-policy
+- [X] remove ingress nginx annotation when `ingress.class` in not nginx
```
Comment (gaetanmaisse) [inline am/CHANGELOG.md:8]:
```suggestion
- [X] Remove ingress nginx annotation when `ingress.class` is not nginx
```

File: am/CHANGELOG.md
Comment (gaetanmaisse) [inline am/CHANGELOG.md:1]: Update on APIM side is missing no?

File: apim/3.x/CHANGELOG.md [hunk 6]
```diff
@@ -3,14 +3,13 @@
...
-
-- [X] Add hook-delete-policy
-
 ### 3.1.65
 
+- [X] Add hook-delete-policy
 - [X] Add check to allow automatic Redis plugin download only for versions prior to 3.21.0 (it is now embedded in the distribution)
 - [X] Add SSL and Sentinel configuration management for Redis and updated doc. 
+- [X] auto generate environment variable `PORTAL_BASE_HREF` and `CONSOLE_BASE_HREF`
+- [X] remove ingress nginx annotation when `ingress.class` in not nginx
```
Comment (gaetanmaisse) [inline apim/3.x/CHANGELOG.md:12]:
```suggestion
- [X] Auto-generate `PORTAL_BASE_HREF` and `CONSOLE_BASE_HREF` environment variables
- [X] Remove ingress nginx annotation when `ingress.class` is not nginx
```

File: apim/3.x/Chart.yaml [hunk 7]
```diff
@@ -20,4 +20,8 @@ annotations:
   # List of changes for the release in artifacthub.io
   # https://artifacthub.io/packages/helm/graviteeio/apim3?modal=changelog
   artifacthub.io/changes: |
-    - Add hook-delete-policy
\ No newline at end of file
+    - Add hook-delete-policy
+    - Add check to allow automatic Redis plugin download only for versions prior to 3.21.0 (it is now embedded in the distribution)
+    - Add SSL and Sentinel configuration management for Redis and updated doc.
+    - auto generate environment variable `PORTAL_BASE_HREF` and `CONSOLE_BASE_HREF`
+    - remove ingress nginx annotation when `ingress.class` in not nginx
```
Comment (gaetanmaisse) [inline apim/3.x/Chart.yaml:27]:
```suggestion
    - Auto-generate `PORTAL_BASE_HREF` and `CONSOLE_BASE_HREF` environment variables
    - Remove ingress nginx annotation when `ingress.class` is not nginx
```

File: apim/3.x/templates/_helpers.tpl (hunk 1 above)
Comment (sebdevaux) [reply apim/3.x/templates/_helpers.tpl:337]: yes, i don't know why but the console work only when CONSOLE_BASE_HREF=/xxxxxxx/ and the portal PORTAL_BASE_HREF=/xxxxx

Comment (gaetanmaisse) [reply apim/3.x/templates/_helpers.tpl:337]: Yeah I was surprised to have the portal working without the `/` 🙈

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**